src/test/java/com/vaadin/flow/ai/formfiller/services/ChatGPTChatCompletionServiceTest.java

```diff
@@ -0,0 +1,141 @@
+package com.vaadin.flow.ai.formfiller.services;
+
+import com.theokanning.openai.completion.chat.ChatCompletionChoice;
+import com.theokanning.openai.completion.chat.ChatCompletionResult;
+import com.theokanning.openai.completion.chat.ChatMessage;
+import com.theokanning.openai.service.OpenAiService;
+import com.vaadin.flow.ai.formfiller.utils.ComponentUtils;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.component.textfield.NumberField;
+import com.vaadin.flow.component.textfield.TextField;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+
+public class ChatGPTChatCompletionServiceTest {
+
+    @Before
+    public void setup() {
+        //set environmental variable
+        System.setProperty("OPENAI_TOKEN", "open-ai-key");
```

Comment:
We can extend `ChatGPTChatCompletionService` and mock methods that actually make calls to AI.
Then we can test `ChatGPTChatCompletionService` implementation with stubs, avoiding calling the AI.
Same for `ChatGPTService`.

Reply:
Thank you, Mikhail!

I think we have  `testGeneratedResponse()` tests both in `ChatGPTService` and `ChatGPTChatCompletionService`, where I mocked the response of the OpenAI API: 
- `OpenAIService.createChatCompletion()` 
- and `OpenAIService.createCompletion()` methods were mocked with real test result/response objects.
(I have added as a comment the result from the endpoints/methods).

So probably these classes are covered properly, but the other suggestions definitely nice and I can fix those!

Reply:
I saw `System.setProperty("OPENAI_TOKEN", "open-ai-key");` and thought that you call AI in the test. If it's not the case, let's resolve and forget this comment. My main concern was about calling AI from unit tests, I think we can avoid it by mocking the service.

Reply:
Now I see, yes, my bad, it was a stub for key. I made so shallow review :)